### DISCUSSION TITLE
[Filament-Android] How to enable and disable one object.

### DISCUSSION THREAD

**Discussion by man4soo** [google/filament#4486] [Q&A]
Hi there.

I'm new to Filament. I have two objects rendered by Filament. I'd like to make one object visible or invisible when a user clicks a button. Each object has a separate material and geometry. How can I do that?

**Comment by romainguy** [comment]
There are two ways: you can simply remove/add the entity from/to the scene, it's very cheap, or use layer masks (you set a mask on both the View and the Renderable, the Renderable is only visible if viewMask & renderableMask != 0)

**Reply by man4soo** [reply]
Thanks for your quick response. I think my explanation wasn't clear. My app has two objects like a big cube and a circumference which is around the cube. They have separate material, and are in the separate geometry, but in the same entity like below. I have only one entity.  I'd like to make the circumference visible/invisible when a user clicks a button. A cube object should be always there. Can you please let me know the way to handle it?


// To create Mesh
val builder = RenderableManager.Builder(2)
      .
     . 
    builder.geometry(0, Cube object vertexes & indexes)
    builder.material(0, cube material)
  
    builder.geometry(1, Circumference object vertexes & indexes)
    builder.material(1, Circumference material)
   .
  .
// After creating Mesh
 mesh?.let {
      filamentView?.scene?.addEntity(it.renderable)
  }

**Reply by romainguy** [reply]
You'll have to separate your object in two entities then. They can use the same parent entity though to make it easy to move them.

**Reply by man4soo** [reply]
Thanks. I'll try it.
